RFC-0001-Agile Coretime.md

```diff
@@ -0,0 +1,291 @@
+
+There is no ability to determine capital requirements for hosting a parachain beyond two years from the point of its initial deployment onto Polkadot. While it would be unreasonable to have perfect and indefinite cost predictions for any real-world platform, not having any clarity whatsoever beyond "market rates" two years hence can be a very off-putting prospect for teams to buy into.
+
+However, quite possibly the most substantial problem is both a perceived and often real high barrier to entry of the Polkadot ecosystem. By forcing innovators to either raise 7-figure sums through investors or appeal to the wider token-holding community, Polkadot makes it essentially difficult for a small band of innovators from deploying their technology into Polkadot. While not being actually permissioned, it is also far from the barrierless, permissionless ideal which an innovation platform such as Polkadot should be striving for.
+
+## Stakeholders
+
+Primary stakeholder sets are:
+
+- Protocol researchers and developers, largely represented by the Polkadot Fellowship and Parity Technologies' Engineering division.
+- Polkadot Parachain teams both present and future, and their users.
+- Polkadot DOT token holders.
+
+_Facilitator:_
+
+Parity Technologies' Ecosystem division.
+
+_Reviewers:_
+
+None.
+
+_Consulted:_
+- Alistair
+- Jonas
+- Bjorn
+- Rob H
+- Rob K
+
+_Socialization:_
+
+This RFC was presented at Polkadot Decoded 2023 Copenhagen on the Main Stage. A small amount of socialisation at the Parachain Summit preceeded it and some substantial discussion followed it. Parity Ecosystem team is currently soliciting views from ecosystem teams who would be key stakeholders.
+
+## Requirements
+
+There are six main requirements:
+
+1. The solution MUST provide an acceptable value-capture mechanism for the Polkadot network.
+2. The solution SHOULD allow parachains and other projects deployed on to the Polkadot UC to make long-term capital expenditure predictions.
+3. The solution SHOULD minimize the barriers to entry in the ecosystem.
+4. The solution SHOULD maximize the value which the Polkadot UC provides by allocating its limited resources optimally.
+5. The design MUST work with a limited set of resources (cores on the Polkadot UC) whose properties and number may evolve over time.
+6. The design MUST avoid creating additional dependency on functionality which the Relay-chain need not strictly provide for the delivery of the Polkadot UC. This includes any dependency on the Relay-chain hosting a DOT token.
+
+Furthermore, the design SHOULD be implementable and deployable in a timely fashion; three months from the acceptance of this RFC would seem reasonable.
```

Comment:
Wouldn't the fellowship be able to whitelist this upgrade?

Reply:
Yes but the track is still 28 days. And even though it is less restrictive to pass earlier on that track, a lot of parachain teams have expressed that they prefer a set block number (i.e. `At` over `After`) for runtime upgrades so that they can prepare for any breaking changes.

Reply:
We'll see. As long as the Relay-chain support for core-assignment exists, then this really shouldn't be a big change.